File: app/src/main/java.backup/com/lumiyaviewer/lumiya/slproto/messages/NearestLandingRegionRequest.java
package com.lumiyaviewer.lumiya.slproto.messages;

import com.lumiyaviewer.lumiya.slproto.SLMessage;
import java.nio.ByteBuffer;

public class NearestLandingRegionRequest extends SLMessage {
    public RequestingRegionData RequestingRegionData_Field = new RequestingRegionData();

    public static class RequestingRegionData {
        public long RegionHandle;
    }

    public NearestLandingRegionRequest() {
        this.zeroCoded = false;
    }

    public int CalcPayloadSize() {
        return 12;
    }

    public void Handle(SLMessageHandler sLMessageHandler) {
        sLMessageHandler.HandleNearestLandingRegionRequest(this);
    }

    public void PackPayload(ByteBuffer byteBuffer) {
        byteBuffer.putShort(-1);
        byteBuffer.put((byte) 0);
        byteBuffer.put((byte) -112);
        packLong(byteBuffer, this.RequestingRegionData_Field.RegionHandle);
    }

    public void UnpackPayload(ByteBuffer byteBuffer) {
        this.RequestingRegionData_Field.RegionHandle = unpackLong(byteBuffer);
    }
}

File: Linkpoint/src/main/app_broken_backup/com/linkpoint/slproto/messages/NearestLandingRegionRequest.kt
package com.linkpoint.slproto.messages

import com.linkpoint.slproto.SLMessage
import java.nio.ByteBuffer

class NearestLandingRegionRequest : SLMessage {
    RequestingRegionData RequestingRegionData_Field = RequestingRegionData()

    class RequestingRegionData {
        Long RegionHandle
    }

    NearestLandingRegionRequest() {
        this.zeroCoded = false
    }

    fun CalcPayloadSize(): Int {
        return 12
    }

    fun Handle(SLMessageHandler sLMessageHandler)  {
        sLMessageHandler.HandleNearestLandingRegionRequest(this)
    }

    fun PackPayload(ByteBuffer byteBuffer)  {
        byteBuffer.putShort(-1)
        byteBuffer.put((Byte) 0)
        byteBuffer.put((Byte) -112)
        packLong(byteBuffer, this.RequestingRegionData_Field.RegionHandle)
    }

    fun UnpackPayload(ByteBuffer byteBuffer)  {
        this.RequestingRegionData_Field.RegionHandle = unpackLong(byteBuffer)
    }
}
